### PR TITLE
APP-26 Update data collection demo according to logging changes

### PR DESCRIPTION
**Pull request by forGGe**

### CODE REVIEW

File: datacollectiondemo/source/c/src/kaa_demo.c [hunk 1]
```diff
@@ -69,6 +71,35 @@ static size_t log_record_counter = 0;
...
+
+/* Under normal conditions this callback shouldn't be called */
+static void timeout_log_delivery(void *context, const kaa_log_bucket_info_t *bucket)
+{
+    (void) context;
+    printf("Timeout reached for log delivery of the bucket: %u!\n", bucket->bucket_id);
+}
+
+static kaa_log_delivery_listener_t log_listener = {
+    success_log_delivery,
```
Comment (rasendubi) [inline datacollectiondemo/source/c/src/kaa_demo.c]: Use designated initializers: `.field = success_log_delivery,`

File: datacollectiondemo/source/c/src/kaa_demo.c [hunk 2]
```diff
@@ -69,6 +71,35 @@ static size_t log_record_counter = 0;
         return (error); \
     }
 
+static void success_log_delivery(void *context, const kaa_log_bucket_info_t *bucket)
+{
+    (void) context;
+    printf("Bucket: %u is successfully delivered. Logs uploaded: %zu\n",
```
Comment (ghost) [inline datacollectiondemo/source/c/src/kaa_demo.c:78]: I suggest to stop the demo after all logs have been received.

File: datacollectiondemo/source/c/src/kaa_demo.c [hunk 3]
```diff
@@ -103,6 +137,10 @@ static void kaa_demo_add_log_record(void *context)
 int main(/*int argc, char *argv[]*/)
 {
     printf("Data collection demo started\n");
+    kaa_log_bucket_constraints_t bucket_sizes = {
+        KAA_DEMO_BUCKET_SIZE,
+        KAA_DEMO_LOGS_IN_BUCKET
```
Comment (rasendubi) [inline datacollectiondemo/source/c/src/kaa_demo.c]: designated initializers?

File: datacollectiondemo/source/c/src/kaa_demo.c [hunk 4]
```diff
@@ -121,9 +159,14 @@ int main(/*int argc, char *argv[]*/)
...
     error_code = kaa_logging_init(kaa_client_get_context(kaa_client)->log_collector
                                 , log_storage_context
-                                , log_upload_strategy_context);
+                                , log_upload_strategy_context
+                                , &bucket_sizes);
+
     KAA_DEMO_RETURN_IF_ERROR(error_code, "Failed to init Kaa log collector");
 
+    error_code = kaa_logging_set_listeners(kaa_client_get_context(kaa_client)->log_collector,
+                                           &log_listener);
```
Comment (rasendubi) [inline datacollectiondemo/source/c/src/kaa_demo.c:178]: `error_code` is not checked

File: datacollectiondemo/source/c/src/kaa_demo.c (hunk 1 above)
Comment (forGGe) [inline datacollectiondemo/source/c/src/kaa_demo.c]: Done

File: datacollectiondemo/source/c/src/kaa_demo.c (hunk 3 above)
Comment (forGGe) [inline datacollectiondemo/source/c/src/kaa_demo.c]: Done

File: datacollectiondemo/source/c/src/kaa_demo.c (hunk 4 above)
Comment (forGGe) [inline datacollectiondemo/source/c/src/kaa_demo.c:178]: Done

File: datacollectiondemo/source/c/src/kaa_demo.c (hunk 2 above)
Comment (forGGe) [inline datacollectiondemo/source/c/src/kaa_demo.c:78]: Done

**Comment by ghost** [comment]
LGTM

**Comment by ohamov** [comment]
LGTM

**Pull request merged**